GCC 4.4 cleaned up some more C++ headers.  You always have to #include
headers directly and cannot rely for things to be included indirectly.

Otherwise you get errors like this:

| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp: In function 'std::string MusicBrainz::intToString(int)':
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp:38:23: error: 'sprintf' was not declared in this scope
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp: In function 'void MusicBrainz::debug(const char*, ...)':
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp:80:10: error: 'stderr' was not declared in this scope
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp:80:33: error: 'fprintf' was not declared in this scope
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/utils_private.cpp:81:26: error: 'vfprintf' was not declared in this scope
| make[2]: *** [src/CMakeFiles/musicbrainz3.dir/utils_private.o] Error 1
| make[2]: *** Waiting for unfinished jobs....
| /home/nitin/build/build0/tmp/work/i586-poky-linux/libmusicbrainz-3.0.2-r0/libmusicbrainz-3.0.2/src/xmlParser/xmlParser.cpp:116:1: warning: 'typedef' was ignored in this declaration
| make[1]: *** [src/CMakeFiles/musicbrainz3.dir/all] Error 2
| make: *** [all] Error 2
| FATAL: oe_runmake failed
NOTE: package libmusicbrainz-3.0.2-r0: task do_compile: Failed

Found the solution here:
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=504895

Date: 2010/07/28
Nitin A Kamble <nitin.a.kamble@intel.com>

Upstream-Status: Inappropriate [debian patch]

Index: libmusicbrainz-3.0.2/src/utils_private.cpp
===================================================================
--- libmusicbrainz-3.0.2.orig/src/utils_private.cpp
+++ libmusicbrainz-3.0.2/src/utils_private.cpp
@@ -20,6 +20,7 @@
  * $Id: utils_private.cpp 9934 2008-07-02 06:57:10Z luks $
  */
  
+#include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <cstdarg>
